### PR TITLE
Feature/turn ending weapons

### PR DESCRIPTION
**Pull request by Matt9440**
#29

### CODE REVIEW

File: code/Pawn/Worm.cs [hunk 1]
```diff
@@ -1,13 +1,17 @@
...
+using TerryForm.States;
 
 namespace TerryForm.Pawn
 {
 	public partial class Worm : Sandbox.Player
 	{
 		[Net] public Weapon EquippedWeapon { get; set; }
 
+		// We should probably compare PlayerID's or something like that rather than comparing the entities directly.
+		public bool IsMyTurn => (Game.StateHandler?.State as PlayingState)?.Turn?.ActivePlayer?.ActiveWorm?.Equals( this ) ?? false;
```
Comment (xezno) [inline code/Pawn/Worm.cs:13]: Not a fan of this, is there no better way to handle it?

File: code/Weapons/Weapon.cs [hunk 2]
```diff
@@ -59,7 +61,19 @@ public override void AttackPrimary()
...
+			if ( !IsFiredTurnEnding )
+				return;
+
+			/* 
+			 * TODO: Let physics resolve and weapon to finish firing before ending the players turn.
+			 * Temporary delay to simulate this.
+			 */
+			await GameTask.DelaySeconds( 1 );
+
+			(Game.StateHandler?.State as PlayingState)?.Turn?.SetTimeRemaining( GameConfig.TurnTimeRemainingAfterFired );
```
Comment (xezno) [inline code/Weapons/Weapon.cs:75]: Not a huge fan of this either, could do with a `CurrentTurn` property or something. Also should `Turn` ever actually be null I think that should be checked before we actually fire the weapon

**Review by xezno — CHANGES_REQUESTED** [review]
Couple of minor things need tweaking, rest looks great

File: code/Pawn/Worm.cs (hunk 1 above)
Comment (Matt9440) [reply code/Pawn/Worm.cs:13]: I'll come back to this one when we figure out the turn stuff.

**Comment by Matt9440** [comment]
I've switched this over to a draft whilst I look into the states/pawns

File: code/Pawn/Worm.cs (hunk 1 above)
Comment (Matt9440) [reply code/Pawn/Worm.cs]: Should be handled now.

**Comment by Matt9440** [comment]
Quite a few changes with a wide scope @xezno

**Review by xezno — APPROVED** [review]
Awesome work, all looks good to me :+1:

**Pull request merged**